src/crc32.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2024 Space Cubics, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <csp/csp.h>
+#include <csp/csp_crc32.h>
+#include <stdio.h>
+#include <endian.h>
+#include <csp/csp_id.h>
+
+struct crc_fixture {
+	csp_packet_t *buf;
+};
+
+static struct crc_fixture crc_fixture;
+
+static void *setup(void)
+{
+	csp_init();
+
+	return &crc_fixture;
+}
+
+static void before(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	fixture->buf = csp_buffer_get(0);
+	zassert_true(fixture->buf != NULL, NULL);
+}
+
+static void after(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	csp_buffer_free(fixture->buf);
+}
+
+ZTEST_F(crc, test_crc_verify_err)
+{
+	csp_packet_t *packet = fixture->buf;
+	int ret;
+
+	/* check error if length < sizeof(crc) */
```

Comment:
Isn't the purpose of this test to check that an error occurs if csp_crc32_append() is not called ?
If the purpose is to create “if length < sizeof(crc)”, the test below can be read as working

```
memcpy(packet->data, "Hello", 5);
packet->length = 5;

ret = csp_crc32_verify(packet);
```

Reply:
rename function name to `test_crc_verify_without_crc`

Reply:
Please update also comment.

`/* check error if length < sizeof(crc) */`